src/states/client.rs

```diff
@@ -37,6 +34,15 @@ use std::{
 /// Duration to wait before sending rate limit exceeded messages.
 pub const RATE_EXCEED_RETRY: Duration = Duration::from_millis(800);
 
+pub struct ClientDetails {
+    pub crust_service: Service,
+    pub full_id: FullId,
+    pub min_section_size: usize,
+    pub msg_expiry_dur: Duration,
+    pub proxy_pub_id: PublicId,
+    pub timer: Timer,
+}
+
 /// A node connecting a user to the network, as opposed to a routing / data storage node.
 ///
 /// Each client has a _proxy_: a node through which all requests are routed.
```

Comment:
Wouldn't it make sense to just use composition to avoid repeating all these fields?